endpoints.go

```diff
@@ -40,6 +42,34 @@ func endpoints(gcfg globalConfig) ([]string, error) {
 	return endpointsFromCluster(gcfg)
 }
 
+func IsLocalEp(ep string) bool {
+	if strings.HasPrefix(ep, "unix:") || strings.HasPrefix(ep, "unixs:") {
+		return true
+	}
+
+	if strings.Contains(ep, "://") {
+		url, err := url.Parse(ep)
+		if err != nil {
+			return false
+		}
+		if url.Scheme == "http" || url.Scheme == "https" {
+			ipStr, _, err := net.SplitHostPort(url.Host)
+			if err != nil {
+				return false
+			}
+			ip := net.ParseIP(ipStr)
+			return ip != nil && ip.IsLoopback()
+		}
+		return false
+	}
+	ipStr, _, err := net.SplitHostPort(ep)
+	if err != nil {
+		return false
+	}
+	ip := net.ParseIP(ipStr)
+	return ip != nil && ip.IsLoopback()
```

Comment:
Have you verified it can successfully detected loopback address in `http://localhost:2379`?